### PR TITLE
Fix mutable lists

### PR DESCRIPTION
**Pull request by mhutch**

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionDataList.cs [hunk 1]
```diff
@@ -251,15 +279,7 @@ public virtual CompletionSelectionStatus FindMatchedEntry (ICompletionDataList c
 					}
 				}
 			}
-			return new CompletionSelectionStatus(idx);
+			return new CompletionSelectionStatus (idx);
 		}
-
-		public virtual int [] GetHighlightedIndices (CompletionData completionData, string completionString)
```
[the remark this answers is not in the document]
Comment (mhutch) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/CompletionDataList.cs]: No it's not, look at line 1957 :)

**Comment by slluis** [comment]
I'm currently refactoring CompletionListWindow and related classes. I will incorporate your changes in my branch (not yet pushed).

**Comment by decriptor** [comment]
@slluis how much with that affect the xml completion?

**Comment by slluis** [comment]
It is an internal refactoring, public api won't change.

**Comment by slluis** [comment]
Fixes included in PR #3446

**Pull request closed without merging**